src/azul/health.py

```diff
@@ -202,3 +227,45 @@ def response(self):
     def generate_cache(self):
         health_object = dict(time=time.time(), health=self.health.as_json())
         self.storage_service.put(object_key=f'health/{self.health.lambda_name}', data=json.dumps(health_object))
+
+    def archive_fail_messages(self, remaining_context_time):
+        client = boto3.client('dynamodb')
+        fail_queue = boto3.resource('sqs').get_queue_by_name(QueueName=config.fail_queue_name)
+        while remaining_context_time.get() > 0:
```

Comment:
You need a margin here. What if you have only 1s left but are entering another 5s blocking call?

Reply:
What do you mean by a blocking call?

Reply:
By margin, you mean something like this?
```suggestion
        while remaining_context_time.get() > 5:
```

Reply:
Kind of. Obviously even 5 is cutting it too close since there is work that you do after you get the messages back.